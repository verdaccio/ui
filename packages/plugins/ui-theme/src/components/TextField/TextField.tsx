import { default as MaterialUITextField, TextFieldProps } from '@mui/material/TextField';
import React, { forwardRef } from 'react';

// The default element type of MUI's TextField is 'div'
type TextFieldRef = HTMLDivElement;

const TextField = forwardRef<TextFieldRef, TextFieldProps>(function TextField(
  { InputProps, classes, ...props },
  ref
) {
  return (
    <MaterialUITextField
      {...props}
      ref={ref}
      InputProps={{
        ...InputProps,
        classes,
      }}
    />
  );
});

export default TextField;
