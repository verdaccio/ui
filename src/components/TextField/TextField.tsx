import { default as MaterialUITextField, TextFieldProps } from '@material-ui/core/TextField';
import React, { forwardRef } from 'react';

// The default element type of MUI's TextField is 'div'
type TextFieldRef = HTMLDivElement;

const TextField = forwardRef<TextFieldRef, TextFieldProps>(function TextField({ InputProps, classes, ...props }, ref) {
  return (
    <MaterialUITextField
      {...props}
      innerRef={ref}
      InputProps={{
        ...InputProps,
        classes,
      }}
    />
  );
});

export default TextField;
