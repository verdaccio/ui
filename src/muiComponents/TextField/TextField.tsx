import React, { forwardRef } from 'react';
import { default as MaterialUITextField, TextFieldProps } from '@material-ui/core/TextField';

// The default element type of MUI's TextField is 'div'
type TextFieldRef = HTMLElementTagNameMap['div'];

const TextField = forwardRef<TextFieldRef, TextFieldProps>(function ToolTip({ InputProps, classes, ...props }, ref) {
  return (
    <MaterialUITextField
      {...props}
      innerRef={ref}
      InputProps={{
        ...InputProps,
        classes,
      }}
    />
  );
});

export default TextField;
