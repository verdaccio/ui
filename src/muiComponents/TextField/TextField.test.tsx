import React from 'react';
import { mount } from 'enzyme';

import TextField from './TextField';

describe('<TextField /> component', () => {
  const props = {
    name: 'test',
    value: 'test',
  };
  test('should render the component in default state', () => {
    const wrapper = mount(<TextField name={props.name} value={props.value} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
