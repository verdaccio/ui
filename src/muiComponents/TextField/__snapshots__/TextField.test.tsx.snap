// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> component should render the component in default state 1`] = `"<div class=\\"MuiFormControl-root MuiTextField-root\\"><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" name=\\"test\\" type=\\"text\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"test\\"></div></div>"`;
